### PR TITLE
feat: frontierWallet support

### PR DESCRIPTION
**Pull request by frontierpawan**
Official Website:
https://www.frontier.xyz/

Chrome Extension Url:
https://chrome.google.com/webstore/detail/frontier-wallet/kppfdiipphfccemcignhifpjkapfbihd

### CODE REVIEW

**Comment by DanielSinclair** [comment]
Hey @frontierpawan. Please QA with the `rainbowkit-example` Vercel deployment here. Many teams have bumped into different behavior between localhost and a production environment.

**Comment by DanielSinclair** [comment]
Bumped into an issue where I am unable to connect the wallet after the initial "Deny" reply upon connect prompt. This may be an issue with the wallet's injector; injection is still required for proper detection in RainbowKit.
Screenshot 2023-02-23 at 1 57 01 AM

**Comment by punithbm** [comment]
Hi @DanielSinclair, thanks for highlighting. Can you please try now. I have attached the screen recording for the reference. Also note we are using manifest version 3 and have observed some issues with new chrome caching behaviour. But I think it should not change anything w.r.t rainbowkit implementation. Please check and let me know.  

https://user-images.githubusercontent.com/13044958/222072677-0a92bdf0-8d87-4c4f-961f-bb706dcaebbf.mov

**Comment by punithbm** [comment]
Hi @DanielSinclair , did you get any chance to look into the above comment ?

**Comment by punithbm** [comment]
@DanielSinclair  Please check this☝️ whenever possible

**Comment by DanielSinclair** [comment]
@punithbm Seeing a CI error. Please run `pnpm run lint:fix`

**Comment by DanielSinclair** [comment]
@punithbm Can't seem to get detection working
Screenshot 2023-03-09 at 9 14 16 PM

**Comment by DanielSinclair** [comment]
Ran CI tests and a deployment to test against.

**Comment by DanielSinclair** [comment]
Please reference the discussion here regarding slow injector issues. This is an issue with the wallet injector, and we are holding off an adopting any API changes in RainbowKit for the time being. XDEFI was able to resolve their issue. https://github.com/rainbow-me/rainbowkit/pull/1042#issuecomment-1503783106

File: packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+
+export interface FrontierWalletOptions {
+  chains: Chain[];
+  shimDisconnect?: boolean;
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts:8]: Deprecate `shimDisconnect` here

File: packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export const frontierWallet = ({
+  chains,
+  shimDisconnect,
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts:13]: Reference recent changes to connectors for `options` pass-through https://github.com/rainbow-me/rainbowkit/blob/0d5640929326d65673596a11cd018a9c6524ff8c/packages/rainbowkit/src/wallets/walletConnectors/metaMaskWallet/metaMaskWallet.ts#L31-L34

File: packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    createConnector: () => ({
+      connector: new InjectedConnector({
+        chains,
+        options: {
+          shimDisconnect,
+          getProvider: () =>
+            //@ts-ignore
+            installed ? (window.frontier?.ethereum as any) : undefined,
+        },
+      }),
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.ts:46]: Missing `qrCode` and `extension` params https://www.rainbowkit.com/docs/custom-wallets

File: packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.svg
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/frontierWallet/frontierWallet.svg:1]: Unrounded icons are prefered as RainbowKit will mask

File: .changeset/short-follow-clap.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@rainbow-me/rainbowkit': minor
+'example': patch
```
Comment (DanielSinclair) [inline .changeset/short-follow-clap.md:3]: Can remove `example` here

File: .changeset/short-follow-clap.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Added support for Frontier wallet
```
Comment (DanielSinclair) [inline .changeset/short-follow-clap.md:6]: Reference a recent changeset with example wallet usage: https://github.com/rainbow-me/rainbowkit/blob/2b4ede4a1e580c3c3a257f4be4fbb5f14ed3beab/.changeset/many-schools-deliver.md?plain=1#L5-L23

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
@frontierpawan Unfortunately am unable to get wallet detection working. This PR is code complete, but will hold-off on merging until that works more reliably. Reference this comment: https://github.com/rainbow-me/rainbowkit/pull/1041#issuecomment-1505573234

**Comment by punithbm** [comment]
@DanielSinclair  We are able to find a workaround to address the injection detection issue in the Rainbow kit. We have submitted the Chrome build for review, and it might take 3-5 days to go live. Once it is live, we will notify you, and you can proceed with retesting it. Thank you for your continued support and the detailed information provided so far.

**Comment by punithbm** [comment]
Hi @DanielSinclair  Our extension update is live.  Can you please test in the latest version of the frontier wallet. Version **1.0.20** is working fine with Rainbowkit. Please check and let us know

**Comment by punithbm** [comment]
Hi @DanielSinclair any update on this ?

**Comment by punithbm** [comment]
Hi @DanielSinclair  Please check and let us know on this PR ? 

cc: @nickbytes

**Comment by punithbm** [comment]
@DanielSinclair  any update on when this can be merged ? Please let us know

**Comment by punithbm** [comment]
@DanielSinclair any update on when this can be merged ? Please let us know

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**